lib/widgets/pin_code_widget.dart

```diff
@@ -141,21 +159,51 @@ class PinCodeWidgetState extends State<PinCodeWidget> {
               ),
             ),
           _numberButton("0"),
-          Container(
-            child: new IconButton(
-              onPressed: () => _setPinCodeInput(_enteredPinCode.substring(0, max(_enteredPinCode.length, 1) - 1)),
-              icon: Icon(
-                Icons.backspace,
-                color: Colors.white,
-              ),
-            ),
-          )
-          ]
-        )
+          widget.onFingerprintEntered == null || ((widget.onFingerprintEntered != null) && _enteredPinCode.length > 0)
+              ? _buildEraseButton()
+              : StreamBuilder<BreezUserModel>(
+                  stream: widget.userProfileBloc.userStream,
+                  builder: (context, snapshot) {
+                    if (!snapshot.hasData || snapshot.data.securityModel.enrolledBiometrics == "") {
+                      return _buildEraseButton();
+                    } else {
+                      if (!_isInit && widget.onFingerprintEntered != null) {
```

Comment:
This should't be called from the build method which should only render widgets. initState is the right place.

Reply:
[Done](https://github.com/breez/breezmobile/pull/210/commits/015c2dfdc9c7eb1f90f158184895e30bc8a3efb2)